crates/noirc_frontend/src/hir_def/mod.rs

```diff
@@ -1,3 +1,20 @@
+//! Noir's Hir is the result of the name resolution step (defined in the
+//! hir module) and is essentially identical to the Ast with some small transformations.
+//! The Hir is the input to the name resolution pass, the type checking pass, and the
+//! monomorphisation pass.
+//!
+//! Name Resolution takes the Ast as input and produces the initial Hir which strongly
```

Comment:
```suggestion
//! The HIR is the input to the name resolution pass, the type checking pass, and the
//! monomorphisation pass.
//!
//! Name Resolution takes the AST as input and produces the initial Hir which strongly
```

Nit but I think these acronyms should be in all caps.